recipes/mdnsresponder/all/conanfile.py

```diff
@@ -1,67 +1,94 @@
-from conans import ConanFile, tools, AutoToolsBuildEnvironment, MSBuild
-from conans.errors import ConanInvalidConfiguration
-from conan.tools.files.symlinks import absolute_to_relative_symlinks
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import symlinks, rmdir, get, replace_in_file, rename, chdir, patch, mkdir, copy, apply_conandata_patches
+from conan.tools.scm import Version
+from conan.tools.gnu import Autotools, AutotoolsToolchain, AutotoolsDeps
+from conan.tools.layout import basic_layout
+from conan.tools.microsoft import MSBuild, is_msvc
 import os
 
-required_conan_version = ">=1.33.0"
+
+required_conan_version = ">=1.53.0"
 
 
 class MdnsResponderConan(ConanFile):
     name = "mdnsresponder"
-    description = "Conan package for Apple's mDNSResponder"
+    description = "Apple Open Source DNS Service Discovery Collection"
     topics = ("Bonjour", "DNS-SD", "mDNS")
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://opensource.apple.com/tarballs/mDNSResponder/"
     license = "Apache-2.0", "BSD-3-Clause"
     settings = "os", "arch", "compiler", "build_type"
     options = {
         "with_opt_patches": [True, False],
+        "use_tls": [True, False],
+        "unicast_disabled": [True, False],
     }
     default_options = {
         "with_opt_patches": False,
+        "use_tls": False,
+        "unicast_disabled": False,
     }
-    exports_sources = ["patches/**"]
 
-    _autotools = None
+    def export_sources(self):
+        # INFO: export_conandata_patches will not export optional patches
+        copy(self, "*.patch", dst=os.path.join(self.export_sources_folder, "patches"), src=os.path.join(self.recipe_folder, "patches"))
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def configure(self):
+        if self.options.use_tls:
+            #The target(libdns_sd) is a shared lib, so mbedtls enable 'shared' option
+            self.options['mbedtls'].shared = True
+
+    def requirements(self):
+        if self.options.use_tls:
+            self.requires("mbedtls/2.25.0")
```

Comment:
```suggestion
            self.requires("mbedtls/3.2.1")
```
Try with latest version available, which has binaries packages in Conan Center.